### PR TITLE
feat(next): create svg css sprites

### PR DESCRIPTION
**Pull request by userquin**
supersedes #235

### CODE REVIEW

**Comment by userquin** [comment]
We'll need to move a few dependencies to dev dependencies to inline `cac`, `colorette`, `consola` and  `unconfig`, I guess it is fine.

I'll try to test both packages using `pnpm pack`  and install `utils` and  `utils-cli` via `file:` protocol.

File: packages/utils-cli/package.json [hunk 1]
```diff
@@ -0,0 +1,52 @@
+{
+	"name": "@iconify/utils-cli",
+	"type": "module",
+	"description": "Common functions for working with Iconify icon sets used by various packages.",
```
Comment (userquin) [inline packages/utils-cli/package.json:4]: Change the description

File: packages/utils-cli/package.json [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+	"author": "Vjacheslav Trushkin",
+	"version": "2.1.7",
+	"license": "MIT",
+	"bugs": "https://github.com/iconify/iconify/issues",
+	"homepage": "https://iconify.design/docs/libraries/utils-cli/",
```
Comment (userquin) [inline packages/utils-cli/package.json:9]: We will need to add the entry in docs website

File: packages/utils/src/svg-css-sprite/types.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+	| AsyncSpriteIconsFactory;
+
+export interface SpriteConfiguration {
+	name: string;
+	collection: SpriteCollection | SpriteCollection[];
+	outdir?: string;
+	mapIconName?: (icon: string, collection?: string) => string;
+}
+
+export interface SpritesConfiguration {
```
Comment (userquin) [inline packages/utils/src/svg-css-sprite/types.ts:42]: Move this to the utils-cli package and expose `defineConfig` function

File: packages/utils/src/svg-css-sprite/create-sprite.ts [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+		options
+	);
+}
+
+export function createReadableStreamSprite(
+	spriteName: string,
+	icons: AsyncSpriteIcons | AsyncSpriteIconsFactory,
+	warn = true
+) {
+	const context: Sprites = { content: '' };
```
Comment (userquin) [inline packages/utils/src/svg-css-sprite/create-sprite.ts:58]: Allow provide context, maybe we'll need to add icons and mappings.

Update other functions here and in nide module.

Comment (userquin) [reply packages/utils/src/svg-css-sprite/create-sprite.ts:58]: Maybe we don't need this, the main problem will be on UnoCSS: when building we'll need to emit the svg sprite (will include a hash) and the css icon url style will need the sprite name.

**Comment by userquin** [comment]
Converting to draft, adding tests...

File: packages/utils/src/svg/parse.ts [hunk 5]
```diff
@@ -0,0 +1,58 @@
+/**
+ * viewBox: x, y, width, height
+ */
+export type SVGViewBox = [number, number, number, number];
```
Comment (userquin) [inline packages/utils/src/svg/parse.ts:4]: `export type SVGViewBox = [x: number, y: number, width: number, height: number];`

Comment (cyberalien) [reply packages/utils/src/svg/parse.ts:4]: Changed

**Comment by cyberalien** [comment]
Closing this PR, see https://github.com/unocss/unocss/pull/2675#issuecomment-1683721033

**Comment by userquin** [comment]
@cyberalien 

We should review this PR and maybe reopen this, we need to do some tests, IIRC you had all the icons in the collections square

**Comment by cyberalien** [comment]
Not sure if this is needed. I'm heavily leaning into SVG+CSS approach instead of sprites. See https://cyberalien.dev/articles/svg-css

It renders very small HTML for icons, while giving full control over icon content: changing stroke width, colors, etc... and allows use of CSS variables to control stuff. So full benefit of CSS + caching and small HTML size.

Earlier this week published experimental Vue and Svelte components that use that technique to render icons.

**Comment by userquin** [comment]
@cyberalien change the favicon, using Nuxt again ;)

imagen

**Pull request closed without merging**